adapters/repos/db/vector/flat/compressed_parallel_iterator.go

```diff
@@ -110,8 +124,15 @@ func (cpi *compressedParallelIterator) IterateAll() chan []BQVecAndID {
 		defer wg.Done()
 		localResults := make([]BQVecAndID, 0, 10_000)
 
+		var vecBuffer []uint64
 		for k, v := c.Seek(seeds[len(seeds)-1]); k != nil; k, v = c.Next() {
-			localResults = append(localResults, extract(k, v))
+			length := len(v) / 8
+			if vecBuffer == nil || len(vecBuffer) < length {
```

Comment:
I think it can be just `if len(vecBuffer) < length {`, right?

Reply:
fixed :)